src/main/java/net/minecraftforge/client/event/OverlayGlintEvent.java

```diff
@@ -0,0 +1,87 @@
+package net.minecraftforge.client.event;
+
+import net.minecraft.client.model.ModelBase;
+import net.minecraft.client.resources.model.IBakedModel;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+/**
+ * Events for hooking into the Glint effect Rendering<br>
+ * Will allow for adding effects to items, creating custom overlays, modifying vanilla
+ * colors, and canceling of glint overlays. 
+ * @author RANKSHANK
+ */
+public class OverlayGlintEvent<T> extends Event
+{
+
+    /**
+     * The model used for the particular glint render.<br><br>
+     * {@link IBakedModel} during {@link StackOverlayEvent}<br>
+     * {@link ModelBase} during {@link ArmorOverlayEvent}<br>
+     * */
+    public final T theModel;
+
```

Comment:
Also it is now becoming the standard to go getters and setters instead of final fields here. It allows for better forward flexibility. So please follow that format.